### PR TITLE
Fixes PLL_Language::get_prop()

### PR DESCRIPTION
**Pull request by Screenfeed**
When using a regex for a composite property name in `PLL_Language::get_prop()`, the matches `1` and `2` should have been used instead of `0` (full match) and `1`.
This PR fixes this issue by using named capturing groups.

This PR also fixes a typo in the value of `PLL_Language::DEPRECATED_URL_PROPERTIES` and adds several tests.

### CODE REVIEW

**Review by Chouby — CHANGES_REQUESTED** [review]

File: tests/phpunit/tests/test-language.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	 * @param WP_UnitTest_Factory $factory
+	 */
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		parent::wpSetUpBeforeClass( $factory );
+		self::create_language( 'en_US' );
+	}
+
+	public static function wpTearDownAfterClass() {
+		remove_action( 'deprecated_property_trigger_error', '__return_false', 581698 );
+	}
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply tests/phpunit/tests/test-language.php]: Removed in [665e5d4](https://github.com/polylang/polylang/pull/1239/commits/665e5d406aa582bb6b01768fba852c0d5aa41aed).

File: tests/phpunit/tests/test-language.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		// Test unknown property.
+		$this->assertFalse( $this->language->get_prop( 'foobar' ) );
+	}
+
+	/**
+	 * PLL_Language::__get()
+	 */
+	public function test___get() {
+		// Test deprecated properties.
+		$did_filter = function_exists( 'did_filter' );
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply tests/phpunit/tests/test-language.php]: I added a polyfill in [1fc8164](https://github.com/polylang/polylang/pull/1239/commits/1fc8164b1ac2bf285296d52681a95859fd2f308c).

File: tests/phpunit/tests/test-language.php
Comment (Hug0-Drelon) [inline tests/phpunit/tests/test-language.php:1]: Should we test `trigger_error()` for deprecated properties as well? With something like `expectDeprecation()`?

Comment (Screenfeed) [reply tests/phpunit/tests/test-language.php:1]: `trigger_error()` doesn't throw exceptions but warnings, so they're not easily catchable, this requires creating a custom error handler, not something I would want to put my fingers in.
Note that we test we're reaching this point with `did_filter()`.

**Pull request merged**